### ISSUE TITLE
Nao compila em OS X 10.11

### ISSUE THREAD

**Issue by mbcrocci** [Plasticoo/CHIP8_C#1]
A versao 1.2 do SDL nao compila bem mesmo depois de mudar include paths, etc. (tem um tipo qualquer no codigo que nao e reconhecido)
Entao exprimentei a SDL 2 que funcionou, mas claro o teu nao tem suporte para o 2.

Sera que mudando so algumas coisas funciona? Qual e a diferenca entre SDL 1.2 e 2?

**Comment by mbcrocci** [comment]
fui mudando o SDL para SDL2 e esta a funcionar ate chegar ao input.c onde nao conhece o SDLKey type, qual e o equivalente em SDL2?

**Comment by Plasticoo** [comment]
Penso que e SDL_SCANCODE_letra

**Comment by mbcrocci** [comment]
esse tipo nao existe
https://wiki.libsdl.org/CategoryAPI

**Comment by Plasticoo** [comment]
Certeza?

Ve isto: https://wiki.libsdl.org/SDL_Scancode

**Comment by Plasticoo** [comment]
Em relacao ao SDL2, ha varias diferencas. Por exemplo, aas instrucoes que precisam de obter as key states sao diferenntes.

**Comment by mbcrocci** [comment]
yup era SDL_Scancode, mas agora esta-se a queixar dessas instrucoes. Eu depois vejo quais sao e mudo

**Comment by mbcrocci** [comment]
o que vale e que sao so algumas
E finalmente percebo o mundo maravilhoso dos makefiles :DD

**Comment by Plasticoo** [comment]
Os makefiles sao fantasticos!!

As instrucoes que talvez precisam de ser mudadas sao: FX0A, EX9E e EXA1.

**Comment by mbcrocci** [comment]
Undefined symbols for architecture x86_64:
  "_SDL_Flip", referenced from:
      _graphics_update in graphics.o
  "_SDL_GetKeyState", referenced from:
      __FX0A in opcodes.o
      __EX9E in opcodes.o
      __EXA1 in opcodes.o
      _input_new_event in input.o
  "_SDL_SetVideoMode", referenced from:
      _graphics_init in graphics.o
  "_SDL_WM_SetCaption", referenced from:
      _graphics_init in graphics.o

**Comment by Plasticoo** [comment]
Foram as tais que te disse. Mas pelos vistos ha mais funcoes diferentes..

**Comment by Plasticoo** [comment]
Tens aqui uma migration guide: https://wiki.libsdl.org/MigrationGuide

**Comment by Plasticoo** [comment]
`SDL_Surface *screen = SDL_SetVideoMode(640, 480, 0, SDL_FULLSCREEN | SDL_OPENGL);`
passa a ser
`SDL_Window *screen = SDL_CreateWindow("My Game Window",
                          SDL_WINDOWPOS_UNDEFINED,
                          SDL_WINDOWPOS_UNDEFINED,
                          640, 480,
                          SDL_WINDOW_FULLSCREEN | SDL_WINDOW_OPENGL);`

**Comment by mbcrocci** [comment]
ok thanks, eu vou vendo isso quando estiver digo-te
